### PR TITLE
=per #18288 Add docs of leveldb queries

### PR DESCRIPTION
**Pull request by patriknw**

### CODE REVIEW

**Comment by patriknw** [comment]
Refs #18288

**Comment by patriknw** [comment]
@ktoso please take a look at this before I replicate it into rst docs.

File: akka-persistence-query/src/main/scala/akka/persistence/query/journal/leveldb/LeveldbReadJournal.scala [hunk 1]
```diff
@@ -22,9 +22,135 @@ import akka.util.ByteString
...
+ *
+ * The stream is completed with failure if there is a failure in executing the query in the
+ * backend journal.
+ *
+ * == AllPersistenceIds ==
+ *
+ * [[akka.persistence.query.AllPersistenceIds]] is used for retrieving all `persistenceIds` of all
+ * persistent actors.
+ *
+ * The order of the returned event stream is unordered and you can expect different order
```
Comment (ktoso) [inline akka-persistence-query/src/main/scala/akka/persistence/query/journal/leveldb/LeveldbReadJournal.scala]: => The returned event stream is unordered

Comment (patriknw) [inline akka-persistence-query/src/main/scala/akka/persistence/query/journal/leveldb/LeveldbReadJournal.scala]: Lol, thx

File: akka-persistence-query/src/main/scala/akka/persistence/query/journal/leveldb/LeveldbReadJournal.scala [hunk 2]
```diff
@@ -22,9 +22,135 @@ import akka.util.ByteString
...
+ *
+ * In addition to the `offset` the `EventEnvelope` also provides `persistenceId` and `sequenceNr`
+ * for each event. The `sequenceNr` is the sequence number for the persistent actor with the
+ * `persistenceId` that persisted the event. The `persistenceId` + `sequenceNr` is an unique
+ * identifier for the event.
+ *
+ * The returned event stream is ordered by the offset (tag sequence number), which corresponds
+ * to the same order as the write journal stored the events. The same stream elements (in same order)
+ * are returned for multiple executions of the query. Deleted events are not deleted from the
+ * tagged event stream.
```
Comment (ktoso) [inline akka-persistence-query/src/main/scala/akka/persistence/query/journal/leveldb/LeveldbReadJournal.scala]: Yeah hm... interesting limitation, maybe for the better to keep so.

Comment (ktoso) [inline akka-persistence-query/src/main/scala/akka/persistence/query/journal/leveldb/LeveldbReadJournal.scala]: Maybe make this bold or stand out somehow in the rst docs: `Deleted events are not deleted from the
+ * tagged event stream.`

**Comment by ktoso** [comment]
LGTM, please rst-ify it :)

File: akka-persistence-query/src/main/scala/akka/persistence/query/journal/leveldb/LeveldbReadJournal.scala (hunk 2 above)
Comment (patriknw) [inline akka-persistence-query/src/main/scala/akka/persistence/query/journal/leveldb/LeveldbReadJournal.scala]: yeah, I found it to be a nice feature (but someone probably wants it differently)

**Comment by patriknw** [comment]
I have pushed the scala version of the rst docs 0c1b30c
Will translate to java.

**Comment by patriknw** [comment]
@ktoso pushed final bits here, now with rst docs for java and scala

**Comment by ktoso** [comment]
Minor comments, should be quickly addressable :)

LGTM

**Comment by patriknw** [comment]
good points, I'll fix them, and squash

**Pull request merged**